### PR TITLE
Add unittests for `tell_trial`

### PR DESCRIPTION
**Pull request by HideakiImamura**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

Nothing.

## What does this implement/fix? Explain your changes.

This PR adds unittests for the `tell_trial` API.

### CODE REVIEW

File: python_tests/test_api.py [hunk 1]
```diff
@@ -399,6 +399,125 @@ def test_delete_study_not_found(self) -> None:
...
+                }
+            ),
+            content_type="application/json",
+        )
+        self.assertEqual(status, 400)
+
+    def test_tell_trial_with_invalid_values(self) -> None:
+        storage = optuna.storages.InMemoryStorage()
+        study = optuna.create_study(storage=storage)
+        for values in [1.0, ["foo"]]:
```
Comment (c-bata) [inline python_tests/test_api.py:503]: nit: How about using `self.subTest()` here?
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

**Review by c-bata — APPROVED** [review]
Looks almost good to me. I left one minor suggestion.

File: python_tests/test_api.py (hunk 1 above)
Comment (HideakiImamura) [reply python_tests/test_api.py:503]: Sounds good. I will use this feature in this PR.

**Comment by HideakiImamura** [comment]
@c-bata Thanks for the review. PTAL.

**Review by c-bata — APPROVED** [review]
LGTM. I'll merge this PR after passing CI.

**Pull request merged**